db/commitlog/commitlog.cc

```diff
@@ -1265,19 +1268,15 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
             background_cycle();
         }
 
-        size_t buf_memory = s;
         if (_buffer.empty()) {
             new_buffer(s);
-            buf_memory += buffer_position();
         }
 
         if (_closed) {
             throw std::runtime_error("commitlog: Cannot add data to a closed segment");
         }
 
-        buf_memory -= permit.release();
-        _segment_manager->account_memory_usage(buf_memory);
-
+        auto pos = buffer_position();
```

Comment:
I don't really understand this code, so pardon my ignorance. Why is this line not at the same place where `size_t buf_memory = s` used to be? Seems like we are skipping some things already written to the buffer here already?

Also, is `s` still used after this?

Reply:
No, the existing code made the mistake of doing `account_memory_usage` using a value based on `s` - i.e. byte size. However, since the underlying buffer is split into sector-sized chunks, each containing overhead for crc, the factual size we need to take into account here is in fact larger. This is handled (below) by instead measuring how much buffer data is use (by comparing resulting position with old). The initial overhead that is included in the old code is instead handled in `new_buffer` (to accommodate `terminate`). 

`s` is not used after (possible) buffer alloc. 

Since we are not interruptible, pushing the `account_memory_usage` makes no difference.